File: Cheat Sheets/Cheat_ Sheet.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    // Basics: Printing and Input
    cout << "Hello, World!" << endl;

    string name;
    cout << "Enter your name: ";
    getline(cin, name);
    cout << "Hello, " << name << endl;

    // Variables and Data Types
    int age = 25;
    double height = 5.9;
    bool isStudent = true;

    // Basic Arithmetic
    int x = 10;
    int y = 5;
    int sumResult = x + y;
    int difference = x - y;
    int product = x * y;
    double division = static_cast<double>(x) / y;
    int remainder = x % y;

    // Vectors and Indexing
    vector<string> fruits = {"apple", "banana", "cherry"};
    string firstFruit = fruits[0];
    fruits.push_back("orange");
    int numFruits = fruits.size();

    // Control Structures: If-Else
    if (age >= 18) {
        cout << "You are an adult." << endl;
    } else {
        cout << "You are a minor." << endl;
    }

    // Loops: For and While
    for (int i = 0; i < 5; ++i) {
        cout << "Iteration " << i << endl;
    }

    int counter = 0;
    while (counter < 5) {
        cout << "While loop iteration " << counter << endl;
        ++counter;
    }

    // Functions
    auto greet = [](const string& name) -> string {
        return "Hello, " + name;
    };
    string message = greet("Alice");

    // Classes and Objects
    class Dog {
    public:
        string name;
        string breed;

        string bark() {
            return name + " barks!";
        }
    };

    Dog myDog;
    myDog.name = "Buddy";
    myDog.breed = "Golden Retriever";
    string dogSound = myDog.bark();

    // Lambda Functions in C++
    auto add = [](int x, int y) { return x + y; };
    int sum = add(3, 4);  // sum is 7

    // Printing Results
    cout << "Variables and Data Types:" << endl;
    cout << age << ", " << height << ", " << isStudent << endl;
    cout << "\nBasic Arithmetic:" << endl;
    cout << sumResult << ", " << difference << ", " << product << ", " << division << ", " << remainder << endl;
    cout << "\nVectors and Indexing:" << endl;
    for (const auto& fruit : fruits) {
        cout << fruit << ", ";
    }
    cout << "\nControl Structures:" << endl;
    cout << "You are an adult." << endl;
    cout << "\nLoops:" << endl;
    cout << "For Loop:" << endl;
    for (int i = 0; i < 5; ++i) {
        cout << "Iteration " << i << endl;
    }
    cout << "While Loop:" << endl;
    counter = 0;
    while (counter < 5) {
        cout << "While loop iteration " << counter << endl;
        ++counter;
    }
    cout << "\nFunctions:" << endl;
    cout << message << endl;
    cout << "\nClasses and Objects:" << endl;
    cout << dogSound << endl;
    cout << "\nLambda Functions in C++:" << endl;
    cout << "Sum: " << sum << endl;

    return 0;
}

File: Cheat Sheets/Cheat_Sheet.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        // Basics: Printing and Input
        System.out.println("Hello, World!");

        System.out.print("Enter your name: ");
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        String name = scanner.nextLine();
        System.out.println("Hello, " + name);

        // Variables and Data Types
        int age = 25;
        double height = 5.9;
        boolean isStudent = true;

        // Basic Arithmetic
        int x = 10;
        int y = 5;
        int sumResult = x + y;
        int difference = x - y;
        int product = x * y;
        double division = (double) x / y;
        int remainder = x % y;

        // Lists and Indexing
        List<String> fruits = new ArrayList<>();
        fruits.add("apple");
        fruits.add("banana");
        fruits.add("cherry");
        String firstFruit = fruits.get(0);
        fruits.add("orange");
        int numFruits = fruits.size();

        // Control Structures: If-Else
        if (age >= 18) {
            System.out.println("You are an adult.");
        } else {
            System.out.println("You are a minor.");
        }

        // Loops: For and While
        for (int i = 0; i < 5; i++) {
            System.out.println("Iteration " + i);
        }

        int counter = 0;
        while (counter < 5) {
            System.out.println("While loop iteration " + counter);
            counter++;
        }

        // Functions
        String message = greet("Alice");

        // Classes and Objects
        class Dog {
            String name;
            String breed;

            String bark() {
                return name + " barks!";
            }
        }

        Dog myDog = new Dog();
        myDog.name = "Buddy";
        myDog.breed = "Golden Retriever";
        String dogSound = myDog.bark();

        // Lambda Functions in Java
        BinaryOperator<Integer> add = (x, y) -> x + y;
        int sum = add.apply(3, 4);  // sum is 7

        // Printing Results
        System.out.println("Variables and Data Types:");
        System.out.println(age + ", " + height + ", " + isStudent);
        System.out.println("\nBasic Arithmetic:");
        System.out.println(sumResult + ", " + difference + ", " + product + ", " + division + ", " + remainder);
        System.out.println("\nLists and Indexing:");
        for (String fruit : fruits) {
            System.out.print(fruit + ", ");
        }
        System.out.println("\nControl Structures:");
        System.out.println("You are an adult.");
        System.out.println("\nLoops:");
        System.out.println("For Loop:");
        for (int i = 0; i < 5; i++) {
            System.out.println("Iteration " + i);
        }
        System.out.println("While Loop:");
        counter = 0;
        while (counter < 5) {
            System.out.println("While loop iteration " + counter);
            counter++;
        }
        System.out.println("\nFunctions:");
        System.out.println(message);
        System.out.println("\nClasses and Objects:");
        System.out.println(dogSound);
        System.out.println("\nLambda Functions in Java:");
        System.out.println("Sum: " + sum);
    }

    // Functions
    static String greet(String name) {
        return "Hello, " + name;
    }
}
